tests/pkg/token/token_utils.go

```diff
@@ -202,3 +204,48 @@ func GetTokenWithClient(t *testing.T, ctx context.Context, client *http.Client,
 
 	return tkn.String()
 }
+
+func FlattenTokenClaims(stdT *testing.T, consumerToken string) string {
+	consumerTokenPayload := getTokenPayload(stdT, consumerToken)
+
+	var jsonMap map[string]interface{}
+	err := json.Unmarshal([]byte(consumerTokenPayload), &jsonMap)
+	require.NoError(stdT, err)
+
+	fm := flatten(jsonMap)
+	claims := "{}"
+	for k, v := range fm {
+		claims, err = sjson.Set(claims, k, v)
+		require.NoError(stdT, err)
+	}
+
+	return claims
+}
+
+func getTokenPayload(t *testing.T, token string) string {
+	// JWT format: <header>.<payload>.<signature>
+	tokenParts := strings.Split(token, ".")
+	require.Equal(t, 3, len(tokenParts), "invalid token format")
+	payload := tokenParts[1]
+
+	b, err := base64.RawURLEncoding.DecodeString(payload)
+	require.NoError(t, err)
+
+	return string(b)
+}
+
+func flatten(m map[string]interface{}) map[string]interface{} {
```

Comment:
Just double check question - have you validated if that func works as expected with some test/dummy input or it is tested properly across the tests?

Reply:
It is used in the newly introduced tests(or the adapted ones) and the ext-svc-mock returns a token that has nested fields that should be validated with this function.